src/ensembl/src/content/app/species/components/species-selection-controls/SpeciesSelectionControls.tsx

```diff
@@ -0,0 +1,147 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { push } from 'connected-react-router';
+import classNames from 'classnames';
+
+import * as urlFor from 'src/shared/helpers/urlHelper';
+
+import { getActiveGenomeId } from 'src/content/app/species/state/general/speciesGeneralSelectors';
+import { getCommittedSpeciesById } from 'src/content/app/species-selector/state/speciesSelectorSelectors';
+
+import {
+  toggleSpeciesUseAndSave,
+  deleteSpeciesAndSave
+} from 'src/content/app/species-selector/state/speciesSelectorActions';
+
+import SlideToggle from 'src/shared/components/slide-toggle/SlideToggle';
+import { PrimaryButton } from 'src/shared/components/button/Button';
+import QuestionButton from 'src/shared/components/question-button/QuestionButton';
+
+import { RootState } from 'src/store';
+
+import styles from './SpeciesSelectionControls.scss';
+
+type SpeciesUseToggle = {
+  isUsed: boolean;
+  onChange: (isUsed: boolean) => void;
+};
+
+type LabelProps = {
+  className: string;
+  onClick?: () => void;
+};
+
+type SpeciesRemovalConfirmationProps = {
+  onConfirm: () => void;
+  onReject: () => void;
+};
+
+const SpeciesSelectionControls = () => {
+  const [isRemoving, setIsRemoving] = useState(false);
+  const genomeId = useSelector(getActiveGenomeId);
+  const species = useSelector((state: RootState) =>
+    getCommittedSpeciesById(state, genomeId || '')
+  );
+  const dispatch = useDispatch();
+
+  if (!genomeId || !species) {
+    return null;
+  }
+
+  const onToggleUse = () => {
+    dispatch(toggleSpeciesUseAndSave(genomeId));
+  };
+
+  const toggleRemovalDialog = () => {
+    setIsRemoving(!isRemoving);
+  };
+
+  const onRemove = () => {
+    dispatch(push(urlFor.speciesSelector()));
+    dispatch(deleteSpeciesAndSave(genomeId));
+  };
+
+  const removeLabelStyles = classNames(styles.remove, styles.clickable);
+
+  return (
+    <div className={styles.speciesSelectionControls}>
+      <SpeciesUseToggle isUsed={species.isEnabled} onChange={onToggleUse} />
+      <div className={styles.removalContainer}>
+        {isRemoving ? (
+          <SpeciesRemovalConfirmation
+            onConfirm={onRemove}
+            onReject={toggleRemovalDialog}
+          />
+        ) : (
+          <span className={removeLabelStyles} onClick={toggleRemovalDialog}>
+            Remove
+          </span>
+        )}
+      </div>
+    </div>
+  );
+};
+
+const speciesUseToggleHelpMessage = `When 'Use' is selected, this species will appear in the species list in all apps.
+'Don't use' will disable this species in other apps, but will not remove it from your list in Species selector.`;
+
+const SpeciesUseToggle = (props: SpeciesUseToggle) => {
+  const doNotUseLabelProps: Partial<LabelProps> = {};
+  const useLabelProps: Partial<LabelProps> = {};
```

Comment:
Can we not make both the fields optional in `LabelProps` instead of using `Partial` here ?